File: server_integration_tests/clients/java_reactive_client/client-app/src/test/java/dev/graphlink/server/Fixtures.java
package dev.graphlink.server;

import dev.graphlink.server.generated.client.DefaultGraphLinkClientAdapter;
import dev.graphlink.server.generated.client.DefaultGraphLinkWebSocketAdapter;
import dev.graphlink.server.generated.client.GraphLinkClient;
import dev.graphlink.server.generated.client.JacksonGraphLinkJsonCodec;

public final class Fixtures {

    private Fixtures() {}

    private static int port() {
        String env = System.getenv("SERVER_PORT");
        try {
            return env != null ? Integer.parseInt(env) : 9997;
        } catch (NumberFormatException e) {
            return 9997;
        }
    }

    public static final String SERVER_URL = "http://localhost:" + port() + "/graphql";
    public static final String WS_URL = "ws://localhost:" + port() + "/graphql";

    public static GraphLinkClient newClient() {
        JacksonGraphLinkJsonCodec codec = new JacksonGraphLinkJsonCodec();
        return new GraphLinkClient(
                new DefaultGraphLinkClientAdapter(SERVER_URL),
                codec, codec,
                new DefaultGraphLinkWebSocketAdapter(WS_URL));
    }
}

File: server_integration_tests/clients/kotlin_client/client-app/src/test/kotlin/dev/graphlink/server/Fixtures.kt
package dev.graphlink.server

import dev.graphlink.server.generated.client.DefaultGraphLinkClientAdapter
import dev.graphlink.server.generated.client.DefaultGraphLinkWebSocketAdapter
import dev.graphlink.server.generated.client.GraphLinkClient
import dev.graphlink.server.generated.client.KotlinxSerializationGraphLinkJsonCodec

object Fixtures {

    private val port: Int
        get() = System.getenv("SERVER_PORT")?.toIntOrNull() ?: 9997

    val SERVER_URL: String get() = "http://localhost:$port/graphql"
    val WS_URL: String get() = "ws://localhost:$port/graphql"

    fun newClient(): GraphLinkClient {
        val codec = KotlinxSerializationGraphLinkJsonCodec()
        return GraphLinkClient(
            DefaultGraphLinkClientAdapter(SERVER_URL),
            DefaultGraphLinkWebSocketAdapter(WS_URL),
            codec, codec,
        )
    }
}
